src/state/error.ts

```diff
@@ -0,0 +1,8 @@
+import { atom } from 'jotai'
+
+export enum ConnectionErrorType {
```

Comment:
Errors may have metadata, so I think it is best to subclass `Error` explicitly:

    class ConnectionError extends Error {}
    class MetaMaskConnectionError extends ConnectionError {}

Reply:
Some prior work:

- https://github.com/Uniswap/widgets/blob/main/src/errors.ts (this can probably be removed from the codebase)
- error codes: https://github.com/Uniswap/widgets/blob/main/src/constants/eip1193.ts
- less relevant: translates known errors to readable strings: https://github.com/Uniswap/widgets/blob/main/src/utils/swapErrorToUserReadableMessage.tsx